### PR TITLE
Add robust handling of RabbitMQ connections

### PR DESCRIPTION
**Pull request by agentultra**
I replace `.connection` method calls on the broker with kombu's own
retry method calls and wrapped them with appropriate exception handlers
and logging messages.

I also added a SystemExit handler to the core loop to control exit from
the loop and handle appropriate exceptions with logging.

When faced with a socket timeout we attempt to re-establish the
connection and will log and exit if that retry mechanism fails to
establish a new connection.

Fixes DHC-2262

### CODE REVIEW

File: akanda/rug/notifications.py [hunk 1]
```diff
@@ -151,7 +157,18 @@ def listen(host_id, amqp_url,
...
+            errback=_handle_connection_error,
+            max_retries=10,
+            interval_start=2,
+            interval_step=2,
+            interval_max=30
+        )
+    except (connection.connection_errors):
+        LOG.exception('Error establishing connection, '
+                      'shutting down...')
+        raise RunTimeError("Error establishing connection to broker.")
```
Comment (ryanpetrello) [inline akanda/rug/notifications.py:171]: Should this be RuntimeError?

File: akanda/rug/notifications.py [hunk 2]
```diff
@@ -151,7 +157,18 @@ def listen(host_id, amqp_url,
         virtual_host=conn_info.path,
         port=conn_info.port,
     )
-    connection.connect()
+    try:
+        connection.ensure_connection(
+            errback=_handle_connection_error,
+            max_retries=10,
```
Comment (drwahl) [inline akanda/rug/notifications.py:163]: This is a bit of a break away from other things in akanda, but we should really start to look at pulling these values from a config file.

File: akanda/rug/notifications.py [hunk 3]
```diff
@@ -135,6 +136,11 @@ def _make_event_from_message(message):
     return event.Event(tenant_id, router_id, crud, message)
 
 
+def _handle_connection_error(exception, interval):
+    LOG.warn("Error establishing connection: %s", exception)
```
Comment (drwahl) [inline akanda/rug/notifications.py:140]: Nit picky, I know, but missing docstrings, even for private functions, make me cringe.

File: akanda/rug/notifications.py (hunk 2 above)
Comment (ryanpetrello) [inline akanda/rug/notifications.py:163]: I agree, though I feel like these are reasonable defaults and better than what we have now (no connection retries).  I'm okay with us making a ticket and adding config options for this later (which also involves updating chef recipes, etc...)

Comment (agentultra) [inline akanda/rug/notifications.py:163]: I was going to suggest we put configuration in a separate PR
